### PR TITLE
Prevent rating before consultation started

### LINKED ISSUE #86: Prevent users from rating a consultation before it's started

### PR DESCRIPTION
**Pull request by DannySS123**
- closes #86

### CODE REVIEW

**Review by kisbogdan-kolos — APPROVED** [review]

File: src/consultations/consultations.controller.ts [hunk 1]
```diff
@@ -235,6 +241,13 @@ export class ConsultationsController {
       )
     }
 
+    if (consultation.startDate > new Date(Date.now())) {
```
Comment (Tschonti) [inline src/consultations/consultations.controller.ts:244]: equivalent to just `new Date()`

File: src/consultations/consultations.service.ts [hunk 2]
```diff
@@ -84,7 +84,28 @@ export class ConsultationsService {
     }))
   }
 
-  async findOne(id: number, user: UserEntity, participationId?: number) {
+  async findOne(id: number, user: UserEntity) {
```
Comment (Tschonti) [inline src/consultations/consultations.service.ts:87]: Emphasize the difference by specifying the return type: `Promise<ConsultationEntity>`

File: src/consultations/consultations.service.ts [hunk 3]
```diff
@@ -84,7 +84,28 @@ export class ConsultationsService {
...
+      )
+    }
+    return consultation[0]
+  }
+
+  async findOneWithRelations(
+    id: number,
+    user: UserEntity,
+    participationId: number,
+  ) {
```
Comment (Tschonti) [inline src/consultations/consultations.service.ts:108]: Emphasize the difference by specifying the return type: `Promise<ConsultationDetails>`

**Review by Tschonti — CHANGES_REQUESTED** [review]
sorry I always find something small :D

**Review by Tschonti — APPROVED** [review]

**Pull request merged**